File: sentimental-mario-less/mario.py
# Define the main function
def main():
    # Get the height of the pyramid from the user
    height = get_height()
    # Build the pyramid using the given height
    build_pyramid(height)


# Define a function to get the height of the pyramid from the user
def get_height():
    # Keep prompting the user until a valid height within the desired range is entered
    while True:
        try:
            # Prompt user for height
            height = int(input("Height: "))
            # Check if the height is within the desired range (1 to 8)
            if 1 <= height <= 8:
                # Return the valid height
                return height
        except ValueError:
            # If the user enters a non-integer value, print an error message
            print("Not an Int")


# Define a function to build the pyramid
def build_pyramid(height):
    # Loop to add a new line for each row of the pyramid
    for i in range(height):
        # Loop to add spaces before the hashes on each row
        for k in range(height - i, 1, -1):
            print(" ", end="")
        # Loop to add hashes on each row
        for j in range(i + 1):
            print("#", end="")
        # Move to the next line after each row is printed
        print()


# Call the main function to start the program
main()

File: mario-less/mario.c
#include <stdio.h>
#include <cs50.h>

void buildPyramid(int height);

int main(void)
{
    // Initialize the variable height
    int height;

    // Run the loop to get a value of height between 1 and 8, from the user
    do
    {
        //Prompt user for height using get_int
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Call the function and pass height to it as a parameter
    buildPyramid(height);
}

// Declare the function buildPyramid
void buildPyramid(int height)
{
    // Loop to add a new line
    for (int i = 0; i < height; i++)
    {
        // Loop to add spaces
        for (int k = height - i; k > 1; k--)
        {
            printf(" ");
        }
        // Loop to add hashes
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}